CONTRIBUTING.md

```diff
@@ -61,3 +62,75 @@ MatcherAssert.assertThat("Some reason", target, matcher);
 MatcherAssert.assertThat("Reasone one", target1, matcher1);
 MatcherAssert.assertThat("Reason two", target2, matcher2);
 ```
+
+## Pull request style
+
+Primary PR rule: it's the responsibility of PR author to bring the changes to the master branch.
+
+Other important mandatory rule - it should reference some ticket. The only exception is a minor type fix in documentation.
+
+Pull request (PR) should consist of two mandatory parts:
+ - "Title" - says **what** is the change, it should be one small and full enough sentence with only neccessary information
+ - "Description" - says **how** this pull requests fixes a problem or implements a new feature
+
+### Title
+
+Title should be as small as possible but provide full enough information to understand what was done (not a process), and where from this sentence.
+It could be capitalized If needed, started from capital letter, and should not include links or references (including tickets numbers).
+
+Good PR titles examples:
+ - Fixed Maven artifact upload - describes what was done: fixed, the what was the fixed: artifact upload, and where: Maven
+ - Implemented GET blobs API for Docker - done: implemented, what: GET blobs API, where: Docker
+ - Added integration test for Maven deploy - done: added, what: integration test for deploy, where: Maven
+
+Bad PR titles:
+ - Fixed NPE - not clear where was the problem, and where; good title could be: "Fixed NPE on Maven artifact download"
+ - Added more tests - too vague; good: "Added unit tests for Foo and Bar classes"
+ - Implementing Docker registry - the process, not the result; good: "Implemented cache layer for Docekr proxy"
+
+### Description
+
+Description starts with a ticket number prefixed with one of these keywors: (Fixed, Closes, For, Part of), then a hiphen a description of the changes. Changes description provides information about **how** the problem from title was fixed. It should be a short summary of all changes to increase readability of changes before looking to code, and provide some context. The format is `(<keyword>For|Closes|Fixes|Part of) #(<ticket>\d+) - (<details>.+)`, e.g.: `For #123 - check if the file exist before accessing it and return 404 code if doesn't`.
```

Comment:
@g4s8 typo in word `hyphen`. And probably to add `and` after `a hyphen`.
Also in example `check if the file exist before accessing` -> `check if the file exists before accessing`